public/tabs/chat/chat_page_content.tsx

```diff
@@ -44,12 +44,27 @@ export const ChatPageContent: React.FC<ChatPageContentProps> = React.memo((props
   const chatActions = useChatActions();
   const registry = getIncontextInsightRegistry();
   const configSchema = getConfigSchema();
+  const latestInputRef = useRef<HTMLDivElement>(null);
+  const [messageSpacerHeight, setMessageSpacerHeight] = useState(0);
 
   useLayoutEffect(() => {
-    if (!props.chatScrollTopRef.current) {
+    if (latestInputRef.current) {
+      setMessageSpacerHeight(window.innerHeight - latestInputRef.current.clientHeight - 135);
+
+      setTimeout(() => {
```

Comment:
Nit: use `requestAnimationFrame`.

Reply:
Great suggestion, just checked that `requestAnimationFrame` is more suitable in this visual update case for a few reasons, thanks!